module/extension/Director/src/Director.cpp

```diff
@@ -213,24 +214,19 @@ namespace module::extension {
             pack_builder.erase(done.requester_task_id);
         });
 
-        // A state that we were monitoring is updated, we might be able to run the reaction now
+        // A state that we were monitoring is updated, we might be able to run the task now
         on<Trigger<StateUpdate>, Sync<Director>>().then("State Updated", [this](const StateUpdate& update) {
-            // TODO ALGORITHM SECTION HERE
-            // This state update event means the conditions for the Provider specified in the update are now met
-        });
-
-        // We do things when we receive tasks to run
-        on<Trigger<TaskPack>, Sync<Director>>().then("Run Task Pack", [this](const TaskPack& pack) {
-            // TODO ALGORITHM SECTION HERE
-            // We have just been given a new task pack
+            // Get the group that had a state update
+            auto p  = providers[update.provider_id];
+            auto& g = groups[p->type];
 
-            // TODO check if there are tasks that were previously in the pack from this Provider and now are gone
-            // For those we need to end execution of those tasks (and their children)
+            // Go check if this state update has changed any of the tasks that are queued
+            reevaluate_queue(g);
+        });
 
-            for (const auto& task : pack) {
-                // TODO run this task through the director's algorithm to work out what to do with it
-                // TODO we need to process this pack as a whole, if one of the required actions fail they all do
-            }
+        // We have a new task pack to run
+        on<Trigger<TaskList>, Sync<Director>>().then("Run Task Pack", [this](const TaskList& pack) {  //
```

Comment:
```suggestion
        on<Trigger<TaskList>, Sync<Director>>().then("Run Task List", [this](const TaskList& list) {  //
```

Reply:
I'm trying to separate the concepts of a task pack and a list of tasks.
It just so happens that they are both just a list of tasks.

I want a task pack to be a set of tasks that are all run together.
While a list of tasks is just a list of tasks.
So the queue of tasks that are waiting to run on a provider group are a list of tasks, but the subtasks that it's trying to run are a task pack.

One you can split up and delete parts, the other should always be taken as a whole